### PR TITLE
Remove KestrelThreadPool abstraction and replaced it with PipeScheduler

### PR DESCRIPTION
**Pull request by davidfowl**
- This should avoid allocations per call to Schedule on .NET Core 2.1 as well.

### CODE REVIEW

**Comment by davidfowl** [comment]
Once we figure out the background crash we'll get rid of these allocations:

image

**Comment by davidfowl** [comment]
OneToTenThreads is the failing test

File: test/Kestrel.Transport.Libuv.Tests/LibuvTransportTests.cs [hunk 1]
```diff
@@ -128,6 +131,12 @@ public class LibuvTransportTests
             }
 
             await transport.UnbindAsync();
+
+            if (!await serviceContext.ConnectionManager.CloseAllConnectionsAsync(default).ConfigureAwait(false))
```
Comment (halter73) [inline test/Kestrel.Transport.Libuv.Tests/LibuvTransportTests.cs]: So this is what was causing test errors?

Comment (davidfowl) [reply test/Kestrel.Transport.Libuv.Tests/LibuvTransportTests.cs]: Yep

**Pull request merged**